heat-stack/app/components/ui/heat/CaseSummaryComponents/EnergyUseHistoryChart.tsx

```diff
@@ -68,48 +68,65 @@ interface EnergyUseHistoryChartProps {
     ) => void;
 }
 
-export function EnergyUseHistoryChart({ lastResult, parsedLastResult, usage_data, recalculateFn }: EnergyUseHistoryChartProps) {
-    const [billingRecords, setBillingRecords] = useState<BillingRecordsSchema>([])
+function objectToString(obj: any): any {
+    // !!!! typeof obj has rules for zodObjects
+    // typeof obj returns the type of the value of that zod object (boolean, object, etc)
+    // JSON.stringify of a Zod object is the value of that Zod object, except for null / undefined
+    if (!obj) {
+        return "none";
+    } else if (typeof obj !== "object") {
+        return JSON.stringify(obj);
+    } else if (Array.isArray(obj)) {
+
+        return `[${obj.map(value => { return objectToString(value) }).join(", ")}]`;
+    }
+
+    const retval = `{ ${Object.entries(obj).map(([key, value]) =>
+        `"${key}": ${objectToString(value)}`).join(", ")} }`;
+    return retval as any;
+}
+
+export function EnergyUseHistoryChart({ lastResult, parsedLastResult, setUsageData, usageData, recalculateFn }: EnergyUseHistoryChartProps) {
+
+
+    const handleOverrideCheckboxChange = (index: number) => {
+        const newRecords = structuredClone(usageData.processed_energy_bills)
+        const period = newRecords[index]
+        
+        if (period) {
+            const currentOverride = period.inclusion_override
+            // Toggle 'inclusion_override'
+            period.inclusion_override = !currentOverride
+            
+            newRecords[index] = { ...period } 
+        }
+        const newUsageData = ({
+            heat_load_output: Object.fromEntries(parsedLastResult?.get('heat_load_output')),
+            balance_point_graph: Object.fromEntries(parsedLastResult?.get('balance_point_graph')),
+            processed_energy_bills: newRecords,
+        }) as UsageDataSchema;
+        setUsageData( (prevUsageData) => {
+
+            if (objectToString(prevUsageData) != objectToString(newUsageData)) {
+                return newUsageData
+            }
+            return prevUsageData // sets useData to
 
-    useEffect(() => {
+        });
         const {
             parsedAndValidatedFormSchema,
             convertedDatesTIWD,
             state_id,
-            county_id} = {...lastResult}
+            county_id } = { ...lastResult }
 
-        console.log('billing records changed, should trigger recalculation: ', billingRecords)
         recalculateFn(
```

Comment:
Surprised you were able to remove the useEffect() but sounds good. It seems to be very reactive.